File: src/07.greedy/02.subArrayGreedy/05.334.go
package sub_array_greedy

import "math"

func IncreasingTriplet(nums []int) bool {
	m1, m2 := math.MaxInt, math.MaxInt

	for _, v := range nums {
		if v <= m1 {
			m1 = v
		} else if v <= m2 {
			m2 = v
		} else {
			return true
		}
	}
	return false
}

File: src/07.greedy/02.subArrayGreedy/05.334.ts
export function increasingTriplet(nums: number[]): boolean {
    
    let m1 = Infinity
    let m2 = Infinity

    for (let i = 0; i < nums.length; i++) {
        if (m1 >= nums[i]) {
            m1 = nums[i]
        } else if (m2 >= nums[i]) {
            m2 = nums[i]
        } else {
            return true
        }
    }

    return false
};